src/Pecee/Http/Input/InputHandler.php

```diff
@@ -128,10 +149,11 @@ public function parseFiles(array $files, ?string $parentKey = null): array
             $keys = [$key];
             $files = $this->rearrangeFile($value['name'], $keys, $value);
 
+            /** @var InputFile[] $list */
             if (isset($list[$key]) === true) {
-                $list[$key][] = $files;
+                $list[$key]->addInputFile(array_values($files));
             } else {
-                $list[$key] = $files;
+                $list[$key] = (new InputFile($key))->setValue(array_values($files));
```

Comment:
We don't have a unittest for this case. I'm not completly sure if it works now after the ca552d5a80e0ccbeb5bcbf7c10dcda459a830acf
@skipperbent